source/manual/test-and-build-a-project-with-github-actions.html.md

```diff
@@ -0,0 +1,266 @@
+---
+owner_slack: "#govuk-developers"
+title: Test & build a project with GitHub Actions
+section: Testing
+layout: manual_layout
+parent: "/manual.html"
+last_reviewed_on: 2020-05-11
+review_in: 6 months
+---
+
+[GitHub Actions](https://github.com/features/actions) is an automated workflow
+system that GOV.UK uses for [Continuous Integration (CI)][ci], we also have
+[Jenkins](/manual/testing-projects.html) and [Concourse](/manual/concourse.html)
+that provide similar functionality. In [GOV.UK RFC 123][] we decided that
+GitHub Actions is the preferred platform for GOV.UK CI usage where the wider
+platform integration of Jenkins is not required.
+
+Please note that we cannot currently make use of the [Secrets][actions-secrets]
+in GitHub Action workflows, there are on-going discussions to resolve this. In
+the meantime please **do not add any secrets to workflows**. For further
+details of this restriction or please contact
+[GOV.UK senior tech](mailto:govuk-senior-tech@digital.cabinet-office.gov.uk).
+
+Deciding the CI platform for a GOV.UK project can be based on the
+following considerations:
+
+- As Jenkins is hosted within GDS' network it can be used to trigger other
+  internal systems to perform jobs during or following a build (for example,
+  triggering [publishing end-to-end tests][e2e-tests] or triggering a
+  deployment)
+- Jenkins can be used for any services that require authentication to external
+  systems (for example, [releasing a gem
+  automatically][gem-release] during a master build), since secrets are not
+  permitted within GitHub Actions
+- As GitHub Actions are not constrained by the [GOV.UK Puppet
+  configuration][govuk-puppet] a wider range of software is available (for
+  example, Node.js versions past 8)
+- As GitHub Actions are not executed within GDS internal network they are a safer
+  choice for testing contributions from external partners (such as pull
+  requests from members of the public)
+
+## GOV.UK Conventions for GitHub Actions
+
+### Name of CI workflow file
+
+You should name the file used for configuring the CI workflow `ci.yml` and name
```

Comment:
Having a consistent name across all of the CI builds Actions is a good thing. It'll make [looking for Actions in repos to enable branch protection for](https://github.com/alphagov/govuk-saas-config/issues/42) easier.